core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -61,8 +66,7 @@ void setSurfaceHolderRequester(SurfaceHolderRequester surfaceHolderRequester) {
 
     void prepareVideo(final Uri videoUri) {
         if (surfaceHolderRequester == null) {
-            logPlayerNotAttachedWarning("prepareVideo()");
-            return;
+            throw new IllegalStateException("Must set a SurfaceHolderRequester before preparing video");
```

Comment:
@joetimmins, @ouchadam applies for the surface holder. 

My let me check was more oriented to the listeners. I was taking a look and it would require a bit of redefinition on how we do things when setting up the facade. 

I am open for giving it a try, what about next PR? I remember @Mecharyry mentioning to me that he wanted to keep this PR as "simply unit testing" as possible